### PR TITLE
[KOGITO-2537] Consume and persist TraceEvents in trusty-service

### PR DESCRIPTION
**Pull request by kostola**
* [Related JIRA ticket](https://issues.redhat.com/browse/KOGITO-2537)
* [Related PR in kogito-apps](https://github.com/kiegroup/kogito-apps/pull/294)

### CODE REVIEW

**Comment by manstis** [comment]
Hi @kostola @r00ta I assume it may be best for me to defer too much work until these are merged?

File: addons/tracing/tracing-decision-api/src/main/java/org/kie/kogito/tracing/decision/event/trace/TraceEvent.java [hunk 1]
```diff
@@ -20,10 +20,13 @@
...
 public class TraceEvent {
 
-    private final TraceHeader header;
-    private final List<TraceInputValue> inputs;
-    private final List<TraceOutputValue> outputs;
-    private final List<TraceExecutionStep> executionSteps;
+    private TraceHeader header;
+    private List<TraceInputValue> inputs;
+    private List<TraceOutputValue> outputs;
+    private List<TraceExecutionStep> executionSteps;
```
[the remark this answers is not in the document]
Comment (kostola) [reply addons/tracing/tracing-decision-api/src/main/java/org/kie/kogito/tracing/decision/event/trace/TraceEvent.java]: Done :+1:

File: addons/tracing/tracing-decision-api/src/main/java/org/kie/kogito/tracing/decision/event/trace/TraceExecutionStep.java [hunk 2]
```diff
@@ -29,21 +30,24 @@
 public class TraceExecutionStep {
```
[the remark this answers is not in the document]
Comment (kostola) [reply addons/tracing/tracing-decision-api/src/main/java/org/kie/kogito/tracing/decision/event/trace/TraceExecutionStep.java:33]: Done :+1:

File: addons/tracing/tracing-decision-api/src/main/java/org/kie/kogito/tracing/decision/event/trace/TraceHeader.java [hunk 3]
```diff
@@ -26,17 +26,20 @@
 
 public class TraceHeader {
```
[the remark this answers is not in the document]
Comment (kostola) [reply addons/tracing/tracing-decision-api/src/main/java/org/kie/kogito/tracing/decision/event/trace/TraceHeader.java:32]: Done :+1:

File: addons/tracing/tracing-decision-api/src/main/java/org/kie/kogito/tracing/decision/event/trace/TraceInputValue.java [hunk 4]
```diff
@@ -28,12 +28,15 @@
 @JsonInclude(NON_NULL)
 public class TraceInputValue {
```
[the remark this answers is not in the document]
Comment (kostola) [reply addons/tracing/tracing-decision-api/src/main/java/org/kie/kogito/tracing/decision/event/trace/TraceInputValue.java:32]: Done :+1:

File: addons/tracing/tracing-decision-api/src/main/java/org/kie/kogito/tracing/decision/event/trace/TraceOutputValue.java [hunk 5]
```diff
@@ -26,13 +26,16 @@
 
 public class TraceOutputValue {
```
[the remark this answers is not in the document]
Comment (kostola) [reply addons/tracing/tracing-decision-api/src/main/java/org/kie/kogito/tracing/decision/event/trace/TraceOutputValue.java:30]: Done :+1:

File: addons/tracing/tracing-decision-api/src/main/java/org/kie/kogito/tracing/decision/event/trace/TraceType.java [hunk 6]
```diff
@@ -23,9 +23,13 @@
 
 @JsonInclude(NON_NULL)
 public class TraceType {
```
[the remark this answers is not in the document]
Comment (kostola) [reply addons/tracing/tracing-decision-api/src/main/java/org/kie/kogito/tracing/decision/event/trace/TraceType.java:28]: Done :+1:

File: addons/tracing/tracing-decision-api/src/main/java/org/kie/kogito/tracing/decision/event/trace/TraceResourceId.java [hunk 7]
```diff
@@ -18,20 +18,20 @@
 
 import com.fasterxml.jackson.annotation.JsonInclude;
 import org.kie.dmn.api.core.DMNModel;
-import org.kie.kogito.tracing.decision.event.evaluate.EvaluateEvent;
 
 import static com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL;
-import static org.kie.kogito.tracing.decision.event.evaluate.EvaluateEventType.AFTER_EVALUATE_DECISION_SERVICE;
-import static org.kie.kogito.tracing.decision.event.evaluate.EvaluateEventType.BEFORE_EVALUATE_DECISION_SERVICE;
 
 public class TraceResourceId {
```
[the remark this answers is not in the document]
Comment (kostola) [reply addons/tracing/tracing-decision-api/src/main/java/org/kie/kogito/tracing/decision/event/trace/TraceResourceId.java:27]: Done :+1:

File: addons/tracing/tracing-decision-api/src/main/java/org/kie/kogito/tracing/decision/event/trace/TraceHeader.java [hunk 8]
```diff
@@ -30,17 +30,20 @@
...
-    private final Long duration;
-    private final TraceResourceId resourceId;
+    private Long duration;
+    private TraceResourceId resourceId;
     @JsonInclude(NON_EMPTY)
-    private final List<Message> messages;
+    private List<Message> messages;
+
+    private TraceHeader() {
+    }
```
[the remark this answers is not in the document]
Comment (kostola) [reply addons/tracing/tracing-decision-api/src/main/java/org/kie/kogito/tracing/decision/event/trace/TraceHeader.java]: Done :+1:

**Comment by kostola** [comment]
> I don't see any related tests for persisting of TraceEvents. Can you please implement some or point me to them if I overlooked it.

@jiripetrlik The only piece of code that persists `TraceEvent`s at the moment is the `trusty-service`, whose code is in the `kogito-app` repository. I don't think we should add any test in `kogito-runtimes` about persistence. Wdyt?

File: addons/tracing/tracing-decision-api/pom.xml [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+    <groupId>org.kie.kogito</groupId>
+    <version>8.0.0-SNAPSHOT</version>
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+
+  <artifactId>kogito-tracing-decision-api</artifactId>
+  <name>Kogito :: Add-Ons :: Tracing :: API</name>
+
+  <properties>
+    <maven.compiler.release>11</maven.compiler.release>
```
Comment (r00ta) [inline addons/tracing/tracing-decision-api/pom.xml:16]: Is this correct?

Comment (kostola) [reply addons/tracing/tracing-decision-api/pom.xml:16]: well..I did the same in the tracing addon so I assume it is

**Comment by kostola** [comment]
jenkins retest this please

**Pull request merged**